### PR TITLE
fix: sc-10414 Increase overscan and update height calculations for better virtual table scrolling experience

### PR DESCRIPTION
**Pull request by khalidwilliams**
This pull request updates GridTable's renderVirtual function to:
1. Calculate element heights with `El.getBoundingClientRect().height` 
2. Add an overscan to crate a smoother scrolling experience 

Previously the ChromeRows on Virtual tables would round their `offsetHeights` somewhat unpredictably when zoomed in, leading to a jumpy scrolling experience. Furthermore `getBoundingClientRect` is the method that virtuoso uses by default for getting an element's height.

### CODE REVIEW

**Review by ksk5280 — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -48,7 +48,7 @@
     "react-router": "^5.2.0",
     "react-router-dom": "^5.2.0",
     "react-stately": "^3.9.0",
-    "react-virtuoso": "^1.9.0",
+    "react-virtuoso": "^2.3.1",
```
Comment (KoltonG) [inline package.json:51]: **question:** Any neat perf updated with the major version change?

File: src/components/Table/GridTable.tsx [hunk 2]
```diff
@@ -544,6 +544,7 @@ function renderVirtual<R extends Kinded>(
   }, [style]);
   return (
     <Virtuoso
+      overscan={5}
```
Comment (KoltonG) [inline src/components/Table/GridTable.tsx:547]: **question:** Is this above and below?

Comment (KoltonG) [reply src/components/Table/GridTable.tsx:547]: Maybe in the future if we need to update this we can expose a prop?

File: src/components/Table/GridTable.tsx [hunk 3]
```diff
@@ -556,12 +557,12 @@ function renderVirtual<R extends Kinded>(
...
         // If it is a chrome row, then we are not using `display: contents;`, return the height of this element.
         if ("chrome" in maybeContentsDiv.dataset) {
-          return maybeContentsDiv.offsetHeight;
+          return maybeContentsDiv.getBoundingClientRect().height || 1;
         }
 
         // Both the `Item` and `itemContent` use `display: contents`, so their height is 0,
         // so instead drill into the 1st real content cell.
-        return (maybeContentsDiv.firstElementChild! as HTMLElement).offsetHeight;
+        return (maybeContentsDiv.firstElementChild! as HTMLElement).getBoundingClientRect().height;
```
Comment (KoltonG) [inline src/components/Table/GridTable.tsx:565]: **question:** Is the `...clientRect` returns a float vs an integer?

**Review by KoltonG — APPROVED** [review]
Looks great! Seems that Chromatic does not see any changes!

**Review by chr1sjf0x — APPROVED** [review]
TY @khalidwilliams 🚀

File: package.json (hunk 1 above)
Comment (khalidwilliams) [reply package.json:51]: No, but the update provided the ability to use their logging / debugging, which was nice.

File: src/components/Table/GridTable.tsx (hunk 2 above)
Comment (khalidwilliams) [reply src/components/Table/GridTable.tsx:547]: I believe it's above and below, but their docs don't use very clear language

File: src/components/Table/GridTable.tsx (hunk 3 above)
Comment (khalidwilliams) [reply src/components/Table/GridTable.tsx:565]: Yep!

File: src/components/Table/GridTable.tsx [hunk 4]
```diff
@@ -556,12 +557,12 @@ function renderVirtual<R extends Kinded>(
 
         // If it is a chrome row, then we are not using `display: contents;`, return the height of this element.
         if ("chrome" in maybeContentsDiv.dataset) {
-          return maybeContentsDiv.offsetHeight;
+          return maybeContentsDiv.getBoundingClientRect().height || 1;
```
Comment (stephenh) [inline src/components/Table/GridTable.tsx:560]: Just curious, but what's better about `getBoundingClientRect()` vs `offsetHeight`? Just more accurate?

Comment (stephenh) [reply src/components/Table/GridTable.tsx:560]: Ah gotcha, gotcha, I read the PR description; makes sense, I think this is great as-is then, i.e. not need for a follow up in-source comment.

**Pull request merged**